orchestrator/process/utils.go

```diff
@@ -25,29 +25,37 @@ func buildStagedProcessesList(managerConfig config.ProcessesManagerConfig) []pro
 	var stagedProcessesList []process
 
 	for _, group := range managerConfig.ProcessGroups {
-		groupArgs := getProcessGroupArgs(group.Script, group.ArgPrefix, group.ArgAssign, group.Args)
+		groupArgs := getProcessArgs(group.Script, group.ArgPrefix, group.ArgAssign, group.Args.GroupArgs)
 		groupInstance := newProcessGroup(group.Name, group.Replicas, group.Command, groupArgs)
 
 		for i := 0; i < group.Replicas; i++ {
-			stagedProcessesList = append(stagedProcessesList, newProcess(groupInstance))
+			processArgs := []string{}
+			if i < len(group.Args.IndividualArgs) {
+				processArgs = getProcessArgs("", group.ArgPrefix, group.ArgAssign, group.Args.IndividualArgs[i])
+			}
+
+			stagedProcessesList = append(stagedProcessesList, newProcess(groupInstance, processArgs))
 		}
 	}
 
 	return stagedProcessesList
 }
 
-// getProcessGroupArgs A function to construct process arguments from config
-func getProcessGroupArgs(script string, argsPrefix string, argsAssign string, args []config.ProcessArg) []string {
-	groupArgs := []string{script}
+// getProcessArgs A function to construct process arguments from config
+func getProcessArgs(script string, argsPrefix string, argsAssign string, args []config.ProcessArg) []string {
```

Comment:
I think we shouldn't fallback to global params in case of individual params not having values. I would pass a bool flag here called `fallback` and check if it's true, then fallback on global params, if not, then skip the param (don't pass a param with empty value).

Reply:
script parameter will be removed from this function

Reply:
I mean the part where we check if `value == ""` and in that case we fallback to global params. This logic shouldn't take place in case of process params. So my suggestion is passing a fallback param and do this check this way `if value == "" && !fallback` then skip this param.

Reply:
This will handle the case where we have this in the yaml file:
```
- Flag: port
```
And no value, or empty value. In this case no `-flag=` shall be passed.